constraint/pkg/regorewriter/regorewriter_test.go

```diff
@@ -91,374 +93,128 @@ func RegoSrc(pkg string, opts ...RegoOption) string {
-			libSrcs: map[string]string{
-				"lib/alpha.rego": RegoSrc("lib.alpha",
-					Body(`
-check(object) {
-  x := data.lib[_]
-  object == "foo"
-}
-`,
-					),
-				),
-			},
-			wantResult: map[string]string{
-				"lib/alpha.rego": `package foo.bar.lib.alpha
-
-check(object) {
-	x := data.foo.bar.lib[_]
-	object == "foo"
-}
-`,
-			},
-		},
-		{
-			name: "entry point references input",
-			baseSrcs: map[string]string{
-				"my_template.rego": RegoSrc("templates.stuff.MyTemplateV1",
-					DenyBody("bucket := input.asset.bucket"),
-				),
-			},
-			wantResult: map[string]string{
-				"my_template.rego": `package templates.stuff.MyTemplateV1
-
-deny[{
-	"msg": message,
-	"details": metadata,
-}] {
-	bucket := input.asset.bucket
-}
-`,
-			},
-		},
-		{
-			name: "lib references input",
-			libSrcs: map[string]string{
-				"lib/my_lib.rego": RegoSrc("lib.myLib",
-					Body(`
-is_foo(name) {
-  input.foo[name]
-}
-`)),
-			},
-			wantResult: map[string]string{
-				"lib/my_lib.rego": `package foo.bar.lib.myLib
-
-is_foo(name) {
-	input.foo[name]
-}
-`,
-			},
-		},
-		{
-			name: "walk data.lib",
-			libSrcs: map[string]string{
-				"lib/my_lib.rego": RegoSrc("lib.myLib",
-					Body(`
-is_foo(name) {
-  walk(data.lib, [p, v])
-}
-`)),
-			},
-			wantResult: map[string]string{
-				"lib/my_lib.rego": `package foo.bar.lib.myLib
-
-is_foo(name) {
-	walk(data.foo.bar.lib, [p, v])
-}
-`,
-			},
-		},
-
-		// Special error cases
-		{
-			name: "lib cannot have package name data.lib",
-			libSrcs: map[string]string{
-				"lib/my_lib.rego": RegoSrc("lib"),
-			},
-			wantError: true,
-		},
-		{
-			name: "lib has invalid package prefix",
-			libSrcs: map[string]string{
-				"lib/my_lib.rego": RegoSrc("mystuff.myLib"),
-			},
-			wantError: true,
-		},
-	}
-
-	for _, tc := range testcases {
-		t.Run(tc.name, tc.Run)
-	}
 }
 
 func TestRegoRewriterErrorCases(t *testing.T) {
 	testcases := []struct {
-		name         string
-		imports      string
-		snippet      string
-		excludeEntry bool
-		excludeLib   bool
+		name      string
+		imports   string
+		snippet   string
+		wantError error
 	}{
 		{
-			name:    "invalid data object reference",
-			snippet: "data.badextern.fungibles[name]",
+			name:      "invalid data object reference",
+			snippet:   "data.badextern.fungibles[name]",
+			wantError: ErrDataReferences,
 		},
 		{
-			name:    "import invalid lib",
-			imports: "data.stuff.foolib",
-			snippet: "foolib.check(input)",
+			name:      "import invalid lib",
+			imports:   "data.stuff.foolib",
+			snippet:   "foolib.check(input)",
+			wantError: ErrInvalidImport,
 		},
 		{
-			name:    "import invalid lib path",
-			imports: "data.stuff.foolib",
-			snippet: "foolib.check(input)",
+			name:      "import invalid lib path",
+			imports:   "data.stuff.foolib",
+			snippet:   "foolib.check(input)",
+			wantError: ErrInvalidImport,
 		},
 		{
 			name: "invalid binding of data to var",
 			snippet: `
-  x := data
-  x.stuff.more.stuff
+ x := data
```

Comment:
nit: can we align the snippet formatting? looks like some start with 2 spaces some with tabs

Reply:
replaced leading spaces with tabs